libs/wingc/src/type_check/symbol_env.rs

```diff
@@ -94,6 +94,11 @@ pub struct SymbolLookupInfo {
 	pub phase: Phase,
 	/// Whether the symbol was defined in an `init`'s environment
 	pub init: bool,
+
+	/// How deep up the symbol environment tree did we look to find this symbol.
+	///
+	/// If this is 0 it means we found it in the entry environment, 1 means first parent 2 next, etc.
+	pub depth: usize,
```

Comment:
Nowhere in this file or in the PR description explains this architectural change / why it was introduced, could you shed some light on this?